archive_converter.py

```diff
@@ -14,65 +14,71 @@
 
 import email
 import os
-from typing import List, Dict
+import re
+from typing import List, Dict, Tuple, Set
 from setup_gmail import Message
+from message_dao import MessageDao
 
 class ArchiveMessageIndex(object):
-    def __init__(self):
-        # Maps message subject (str) to starting message of email thread
-        self._key_to_thread_start : Dict[str, Message] = {}
-        # Maps message.id to message
-        self._messages_seen : Dict[str, Message] = {}
-
-    def find_thread(self, key: str) -> Message:
-        return self._key_to_thread_start[key]
-
-    def update(self, data_dir: str):
+    def __init__(self, message_dao : MessageDao):
+        self._message_dao = message_dao
+        
+    def update(self, data_dir: str) -> Tuple[Tuple[Message, ...], Tuple[Message, ...]]:
```

Comment:
this should probably now have a docstring given it has a not-too-obvious return type

Reply:
> Optional: also split out the Dao refactor into a parent commit and explain in the commit message there a bit about the rationale (i.e. that the change doesn't have behavior now but is intended to let us use Postgres in the future)

How would I go about doing this?

Reply:
idk why this showed up here, but I meant to ask this in reply to the review message.